docs/spec/index.md

```diff
@@ -1,54 +1,67 @@
-| ops_location_desc | String | Optional | Description of the operational location. |
-| ops_location_lat | Latitude | Required | The latitude of the operational location. |
-| ops_location_lon | Longitude | Required | The longitude of the operational location. |
+| **File Name** | **Type** | **Description** |
+| --- | --- | --- |
+| trips_supplement.txt | Supplement | Supplements and modifies GTFS [trips.txt](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#tripstxt) with deadheads and other non-public trip information.|
+| stops_supplement.txt | Supplement | Supplements and modifies GTFS [stops.txt](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#stopstxt) with internal stop locations, waypoints, and other non-public stop information.|
+| stop_times_supplement.txt | Supplement | Supplements and modifies GTFS [stop_times.txt](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#stop_timestxt) with non-public times that trips stop at locations, and related information. |
+| routes.txt | Supplement | Supplements and modifies GTFS [routes.txt](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#routestxt) with internal route identifiers and other non-public route identification. |
+| runs_pieces.txt | TODS-Specific | Defines daily personnel schedules within a feed. |
+| run_events.txt | TODS-Specific| Defines other scheduled activities to be performed by a member of personnel during a run. |
 
-### deadhead_times.txt
+The use of the `_supplement` standard to modify other GTFS files remains experimental at this time.
+
+
+## `_supplement` Files
+
+### Structure & Evaluation
+
+The fields in all `_supplement` files match those defined in the corresponding file's [GTFS specification](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md).
+
+Each row in a `_supplement` file is used to modify the corresponding file in GTFS and should be evaluated as follows:
+
+1. If the row's Primary Key is defined in the corresponding GTFS file, and the `TODS_delete` field is equal to `1`, remove the row from the GTFS file.
+2. If the row's Primary Key is defined in the corresponding GTFS file, and the `TODS_delete` field is not defined or not equal to `1`, set or update (in the corresponding GTFS file) the value of any non-blank fields defined in the `_supplement` file's row to the `_supplement`'s value.
+3. If the row's Primary key is NOT defined in the corresponding GTFS file, add the row to the corresponding GTFS file.
+
+In other words, where a primary key matches, the row is either removed or any non-empty values in the row are used to *update* the corresponding GTFS values. Where a primary key match does not exist, the entire row is added.
+
+> _**Implications and Guidance:**_
+> - To explicitely blank a field, delete the row entirely using the `TODS_delete` field and then re-add the row with the desired values subsequent thereto.
```

Comment:
If the only way to remove a single column is to delete the row and recreate it, that seems like an oversight in the spec. Can we fix it?

Some specific reasons that's bad:
- It's clunky to write two lines instead of one. Especially if you need to delete an entire column for all rows. The file would end up being twice as long as the GTFS it amends. 
- The primary key would then show up twice in this file. If the order is wrong, then you could delete the row you just modified. GTFS files should have unique primary keys and be order-independent.
- Requires duplicating any columns that you want to save unmodified from GTFS, instead of leaving the field blank.

Two ideas for a solution:
1. Any GTFS column could be given a value of "ODS_blank" (and hope nobody every has a stop with exactly that name)
2. Instead of `ODS_delete` being a simple boolean, it's a list of columns to delete

Reply:
I personally don't see a use case for blanking a field, since it's not the case where an operator might want to obscure internal data — this is a spec for adding non-public information to public information; anything extraneous need not be removed from the existing feed, but could simply ignore it in an internal consumer application.

The concept of replicating an entire row seems like a non-starter to me. At that point, we're basically back to publishing internal and external GTFS feeds, and many — myself included — would simply just follow that practice if I already needed to worry about validating each specific field in two files.

Does anyone have a specific use case for why they'd need to delete a column value (without replacing it)? Otherwise, I'm inclined to say we should ignore for now.

Reply:
(The only reason I added this note is from a discussion with an industry colleague doing a once-over who made the observation. I do like the premise of having things processed in-order to ensure things can be properly replicated. Or to preserve order-independence, per Sky's comment, should we explicitly prohibit this behavior and say a primary key that's already used in GTFS can't be deleted and re-added via TODS?)

Reply:
I think Primary Keys should be unique in these files, and we shouldn't allow a duplicate row even if one is to delete.

I don't have a specific use case for blanking fields, but my hypothetical use case isn't about obscuring data, but about removing inaccurate data. Hypothetically, maybe an agency schedules bus stop times only at timepoints, and adds interpolated times at other stops to the public GTFS, but doesn't want to have the interpolated times in the internal feed.

I'm fine with not allowing blanking a field now, as long as we think we can solve it in a backwards-compatible way if it does come up later.

I looked through the GTFS docs, and didn't see any places we'd end up with an invalid GTFS file if we couldn't blank fields, even if we expand to other GTFS files. All optional enums have a value that's equivalent to blank, and all conditionally forbidden fields wouldn't have a way to become forbidden by adding TODS files.

Reply:
➕ Agreed with @skyqrose on further review, it's probably not worth trying to jam a solution right now; the other strongest example I could come up with is changing a route designation internally, and wanting to nullify a `route_long_name` or `route_short_name`, but it might be more likely you'd just move to use a new `route_id` anyway where you could dictate that.